cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtils.java

```diff
@@ -144,4 +146,46 @@ public static void updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus, Kafka
             }
         }
     }
+
+    /**
+     * Updates the {@code KafkaRebalanceStatus} with a new Warning condition based
+     * on the provided {@link Throwable}. If a Warning condition with the same reason
+     * and message already exists, no update is performed.
+     *
+     * @param status    The {@link KafkaRebalanceStatus} object whose conditions will be updated.
+     * @param exception The {@link Throwable} containing the reason and message for the Warning condition.
+     */
+    public static void addWarningCondition(KafkaRebalanceStatus status, Throwable exception) {
+        List<Condition> conditions = status.getConditions() != null ? new ArrayList<>(status.getConditions()) : new ArrayList<>();
+
+        String causeDescription = exception.getCause() != null
+                ? exception.getCause().toString()
+                : "No cause available";
+        Condition newCondition = StatusUtils.buildWarningCondition(causeDescription, exception.getMessage());
+        Condition oldCondition = conditions.stream()
+                .filter(c -> "Warning".equals(c.getType()))
+                .findFirst()
```

Comment:
What if there is more than one warning?

Reply:
We update the list with the latest Warning unless the existing Warning has the same reason and message. We want to keep the original Warning of the same reason and message to track so we know when it first occurred.